File: 15 exercicios POO/4/golang/main.go
package main
import "fmt"

type Animal struct{
    especie string
    nome string
}
func (a Animal) EmitirSom() string{
    return ""
}

type Cachorro struct{
    Animal
}

func (c Cachorro) EmitirSom() string {
    return "AU AU"
}

type Gato struct {
    Animal
}

func (g Gato) EmitirSom() string {
    return "MIAU"
}

func main() {
    cachorro := Cachorro{Animal{"cachorro", "chorro"}}
    fmt.Println(cachorro.EmitirSom()) // AU AU

    gato := Gato{Animal{"gato", "miau"}}
    fmt.Println(gato.EmitirSom()) // MIAU
}

File: 15 exercicios POO/4/java/src/Main.java
public class Main {
    public static void main(String[] args) {
        Cachorro cachorro = new Cachorro();
        cachorro.especie = "cachorro";
        cachorro.nome = "chorro";
        cachorro.emitir_som();

        Gato gato = new Gato();
        gato.especie = "gato";
        gato.nome = "to";
        gato.emitir_som();
    }
}